vispy/visuals/text/text.py

```diff
@@ -438,7 +438,10 @@ def pos(self):
     @pos.setter
     def pos(self, pos):
         pos = [float(p) for p in pos]
-        assert len(pos) == 2
+        l = len(pos);
+        assert(l == 3 or l == 2)
```

Comment:
`assert` statements are skipped when optimization is used, do a proper check and throw an error (see other instances in code base)